### ISSUE TITLE
MA0182 false positive on `[DynamicallyAccessedMembers]`

### ISSUE THREAD

**Issue by dmitriy-maksimov-wd** [meziantou/Meziantou.Analyzer#962]
### Version of the Meziantou.Analyzer NuGet package

2.0.275

### Rule Identifier

MA0182

### Target Framework

net10.0

### C# Language version

C# 12

### Description

`MA0182` report the follow error

```
21>LocalTaskHandlerTypeUnitTests.cs(31,23): Error MA0182 : Internal type 'FakeTaskHandler1' is apparently never used. If so, remove it from the assembly. If this type is intended to contain only static members, make it 'static'. (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0182.md)
```

but it marked with `DynamicallyAccessedMembers` attribute
```
	[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
	private sealed class FakeTaskHandler1 : ITaskHandler
```

According to [docs](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-10.0) 
> Indicates that certain members on a specified Type are accessed dynamically, for example, through Reflection.

### Reproduction Steps

Unfortunately, I can't reproduce the issue on a sample project

**Comment by meziantou** [comment]
Thanks for reporting the bug! Should be fixed in the latest version
